scripts/lint_jjb.py

```diff
@@ -155,6 +168,149 @@ def check_retention(job, in_file):
     return return_value
 
 
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+def check_timed_trigger_value(trigger):
+
+    # Non-standard CRON values allowed by jenkins
+    # Reference:
+    # https://en.wikipedia.org/wiki/Cron#Nonstandard_predefined_scheduling_definitions
+    # http://www.scmgalaxy.com/tutorials/setting-up-the-cron-jobs-in-jenkins-using-build-periodically-scheduling-the-jenins-job/
+    allowed_non_standard_values = [
+        "@yearly", "@annually", "@monthly", "@weekly", "@daily",
+        "@midnight", "@hourly"
+    ]
+
+    # If the trigger provided is a NoneType,
+    # then the job has no value, so pass it.
+    if trigger is None:
```

Comment:
I don't think so, and even if it did would we want jobs which set it to that?

Reply:
I don't know, I guess it depends what it would mean. The code sets `trigger` to `None` if it's not present. My question is so I can understand whether or not the code needs to distinguish between a null value created by this code, due to the absence of a value, and a null value from the JJB.

Reply:
To be perfectly honest I can't think of a way to distinguish the difference. I'm not entirely sure it'd be worth it either, because to my mind I don't think we'd want any jobs to be configured with a null value in the JJB.

I can say that if an empty string is passed to croniter, it fails it:

```
>>> from croniter import croniter
>>> croniter.is_valid("")
False
```

Reply:
At the moment this code does allow it to be set to null. I guess if it becomes an issue we can always look at it then.